src/RankMovie.jsx

```diff
@@ -38,11 +53,46 @@ function RankMovie() {
     }
   }, [low, high, newMovie]);
 
-  const handleComparison = (prefersNewMovie) => {
-    if (prefersNewMovie) {
-      setLow(mid + 1);
-    } else {
-      setHigh(mid - 1);
+  function updateMovieRatings(movieId, rating) {
+    const userId = currentUser.uid;
+    fetch("http://localhost:3000/users/logMovie", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ userId, movieId, rating }),
+    })
+      .then((response) => response.json())
+      .catch((error) => {
+        console.error("Error logging movie:", error);
+      });
+  }
+
+  const handleComparison = async (prefersNewMovie) => {
+    const winner = prefersNewMovie ? newUserMovie : userMovies[mid];
+    const loser = prefersNewMovie ? userMovies[mid] : newUserMovie;
+
+    const K = 1;
+    const divisor = 100;
+    const winnerExpected =
+      1 / (1 + Math.pow(10, (loser.rating - winner.rating) / divisor));
```

Comment:
Could you share how you came up with this formula? In its current state the formula is $\frac{1}{1 + 10^{(L-W)/10}}$. The $Max(L-W)$ would be 9 and and the minimum would be -9. This means the range `winnerExpected` can take on is between $(.45,.55)$. I think this will cause trouble getting the movie ratings to converge, unless you find a way to dynamically adjust K.

Reply:
I think a dynamically adjusted K based on number of comparisons completed could potentially solve the problem(the more comparisons we have done the more certain we are about a movies rating so K factor will go down). But then there is another problem that is present: If i say a movie is bad, its initial rating will be set to 2.5. If I then select this new "bad" movie as the winner in all subsequent comparisons, then the movie should theoretically be the best movie and have the highest ranking, but it will most likely not have the highest rating since each matchup only increases the rating by around 0.5. In this scenario I would want the K factor to go up rather than down. Do you have any suggestions to combat this?

Reply:
Note: even if I set all movies initial ratings to a generic 5/10, the same problem would still persist